operations/observability/Observability.md

```diff
@@ -0,0 +1,300 @@
+
+3. Boot, install, and upgrade duration monitoring.
+   This will do automatic calculation and reporting of both the time a given section of install/upgrade has taken, as well as metrics on how many GOSS tests results show OK versus PROBLEM status.
+
+4. Time, node, product stream, capacity, and other dimension based health and performance insight.
+
+5. Multi-Interval Continuous Health Checks.
+   Define and implement regular automatic scheduled health checks to occur both during install and upgrade, as well as after the install or upgrade has been completed.
+   The frequency needs to be determined but will likely include every six hours, once a day, and once a week.
+
+6. Automate the deployment, installation, and configuration of the Install and Upgrade Observability framework in a CSM environment.
+
+7. The automatic generation of configurable Grafana dashboards that provide key insights and KPIs into the frequency of errors across the complex systems, panels to visualize the outliers, and trends in the complex system  across different dimensions.
+
+The following IUF topics are discussed in the following subsections.
+
+- [Automation of Observability framework](#automation-of-observability-framework)
+  - [Features](#automation-framework-features)
+  - [Automation workflow](#automation-workflow)
+  - [`systemd` services](#systemd-services)
+- [IUF Timing dashboard](#iuf-timing-dashboard)
+  - [Features](#timing-dashboard-features)
+  - [Prometheus metrics using Argo Workflow](#prometheus-metrics-using-argo-workflow)
+  - [Timing Dashboard](#timing-dashboard)
+- [GOSS Tests for PIT and NCN](#goss-tests-for-pit-and-ncn)
+  - [Overview](#goss-test-overview)
+  - [Workflow](#workflow)
+  - [Log file format](#log-file-format)
+  - [Grok-exporter deployment, service and service-monitor](#grok-exporter-deployment-service-and-service-monitor)
+  - [Configuration file for the grok-exporter](#configuration-file-for-the-grok-exporter)
+  - [Prometheus metrics and Grafana dashboard](#prometheus-metrics-and-grafana-dashboard)
+- [Error Dashboard](#error-dashboard)
+  - [Features](#error-dashboards-features)
+  - [Error Dashboards](#error-dashboards)
+
+## Automation of Observability framework
+
+Grok-exporter, Prometheus, and Grafana get instantiated automatically on the Combined Install Media LiveCD for PIT dimension of metrics for error/debug/count.
+
+### Automation framework features
+
+- Prometheus, grok-exporter, and Grafana containers deployed initially on the `csm-pit` remote server for monitoring before systems are installed.
+- Define and build IaC repo to store observability configuration.
+- Define and build automations to update configuration of observability configuration when the IaC configuration repo is updated.
+- Create an RPM with all the three containers and services for `csm-pit` and `csm-shared-pit` using Jenkins pipeline.
```

Comment:
This really sounds like it's intended for internal use, not customers. Or am I way offtrack? Why would this be in the public CSM docs?

Reply:
Changed